contracts/rewards/AutoCompoundingRewards.sol

```diff
@@ -305,10 +322,51 @@ contract AutoCompoundingRewards is IAutoCompoundingRewards, ReentrancyGuardUpgra
         emit ProgramEnabled({ pool: pool, status: status, remainingRewards: p.remainingRewards });
     }
 
+    /**
+     * @inheritdoc IAutoCompoundingRewards
+     */
+    function setAutoTriggerCount(uint256 newAutoTriggerCount) external onlyAdmin {
+        uint256 prevAutoTriggerCount = _autoTriggerCount;
+        if (prevAutoTriggerCount == newAutoTriggerCount) {
+            return;
+        }
+
+        _autoTriggerCount = newAutoTriggerCount;
+
+        emit AutoTriggerCountUpdated({
+            prevAutoTriggerCount: prevAutoTriggerCount,
+            newAutoTriggerCount: newAutoTriggerCount
+        });
+    }
+
+    /**
+     * @inheritdoc IAutoCompoundingRewards
+     */
+    function trigger() external nonReentrant {
+        address[] memory values = _pools.values();
+        uint256 numOfPools = values.length;
+        uint256 startIndex = _nextTriggerIndex;
+        uint256 count = Math.min(numOfPools, _autoTriggerCount);
+
+        for (uint256 i = 0; i < count; i++) {
+            uint256 index = (startIndex + i) % numOfPools;
+            _processRewards(Token(values[index]), true);
+        }
+
+        _nextTriggerIndex = (startIndex + count) % numOfPools;
+    }
+
     /**
      * @inheritdoc IAutoCompoundingRewards
      */
     function processRewards(Token pool) external nonReentrant {
+        _processRewards(pool, false);
+    }
+
+    /**
+     * @dev processes the rewards of a given pool
+     */
+    function _processRewards(Token pool, bool skip) private {
```

Comment:
skip implies that if true, the request is going to be skipped

Reply:
fixed